### ISSUE TITLE
AttributeError: 'WebRequest' object has no attribute 'channel'

### ISSUE THREAD

**Issue by brianmay** [django/daphne#178]
When testing the fix for https://github.com/django/channels/issues/897 I noticed this error between tests:

```
[2018-02-21 09:16:54,587: ERROR/asyncio] Exception in callback AsyncioSelectorReactor.callLater.<locals>.run() at /home/brian/.pyenv/versions/3.6.4/envs/myrocc-channels2/lib/python3.6/site-packages/twisted/internet/asyncioreactor.py:287
handle: <TimerHandle when=245658.386868275 AsyncioSelectorReactor.callLater.<locals>.run() at /home/brian/.pyenv/versions/3.6.4/envs/myrocc-channels2/lib/python3.6/site-packages/twisted/internet/asyncioreactor.py:287>
Traceback (most recent call last):
  File "/home/brian/.pyenv/versions/3.6.4/lib/python3.6/asyncio/events.py", line 145, in _run
    self._callback(*self._args)
  File "/home/brian/.pyenv/versions/3.6.4/envs/myrocc-channels2/lib/python3.6/site-packages/twisted/internet/asyncioreactor.py", line 290, in run
    f(*args, **kwargs)
  File "/home/brian/tree/myro/channels2/daphne/daphne/server.py", line 274, in timeout_checker
    protocol.check_timeouts()
  File "/home/brian/tree/myro/channels2/daphne/daphne/http_protocol.py", line 248, in check_timeouts
    self.basic_error(503, b"Service Unavailable", "Application failed to respond within time limit.")
  File "/home/brian/tree/myro/channels2/daphne/daphne/http_protocol.py", line 281, in basic_error
    (b"Content-Type", b"text/html; charset=utf-8"),
  File "/home/brian/tree/myro/channels2/daphne/daphne/http_protocol.py", line 193, in handle_reply
    if self.channel is None:
AttributeError: 'WebRequest' object has no attribute 'channel'
```

I am going to try the following change for now:

```diff
diff --git a/daphne/http_protocol.py b/daphne/http_protocol.py
index 7044109..0e03491 100755
--- a/daphne/http_protocol.py
+++ b/daphne/http_protocol.py
@@ -190,7 +190,7 @@ class WebRequest(http.Request):
         Handles a reply from the client
         """
         # Handle connections that are already closed
-        if self.channel is None:
+        if getattr(self, 'channel', None) is None:
             return
         # Check message validity
         if "type" not in message:
```

**Comment by andrewgodwin** [comment]
Hmm, that's weird, not sure how it doesn't exist yet but that change seems appropriate.

**Comment by brianmay** [comment]
I am no twisted expert, but I suspect `_cleanup` is being called when the connection is closed, and hence this line is deleting the `self.channel` object: https://github.com/twisted/twisted/blob/trunk/src/twisted/web/http.py#L750

**Comment by andrewgodwin** [comment]
This should be fixed by 0a2c6c2ff230a5d65706de5a3cc8e2670cf9d71c - please reopen if you can still repro it after that!
